source/common/router/shadow_writer_impl.cc

```diff
@@ -13,15 +13,21 @@ namespace Router {
 
 void ShadowWriterImpl::shadow(const std::string& cluster, Http::MessagePtr&& request,
                               std::chrono::milliseconds timeout) {
+  // It's possible that the cluster specified in the route configuration no longer exists due
+  // to a CDS removal. Check that it still exists before shadowing.
+  // TODO(mattklein123): Optimally we would have a stat but for now just fix the crashing issue.
+  if (!cm_.get(cluster)) {
+    return;
```

Comment:
I wonder if we should at least leave a warn log behind so people are not at a loss for why they're not seeing shadow traffic.

Reply:
I think a warn log is not a great idea since this is in the data path and could spam quite a bit. Though I had considered at least adding a debug log so I will do that.

Reply:
As you mention in the comment, I think a stat is most appropriate in this path. But I'm fine with merging this crash-fix as-is. @junr03 you ok with that?